vispy/visuals/_scalable_textures.py

```diff
@@ -359,18 +397,20 @@ def _get_texture_format_for_data(self, data, internalformat):
             num_channels = self._data_num_channels(data)
             texture_format = self._handle_auto_texture_format(internalformat, data)
             texture_format = self._get_gl_tex_format(texture_format, num_channels)
+        else:
+            raise NotImplementedError("wut?")
```

Comment:
This seems to be an oversight. Any idea what to do here @djhoese? Maybe internfalformat should simply not be None?

Reply:
Yeah I guess this is a result of this code being extracted from the original visual modules/logic. Logically, this texture subclass should never be used if `texture_format` (passed to this class as `internalformat` is `None`. So maybe start the method with an `if internalformat is None: raise ValueError(...)`?

Reply:
Done!